validator/src/journal/block_manager.rs

```diff
@@ -398,13 +402,152 @@ impl BlockManagerState {
 #[derive(Default, Clone)]
 pub struct BlockManager {
     state: Arc<BlockManagerState>,
+    persist_lock: Arc<RwLock<()>>,
 }
 
 impl BlockManager {
     pub fn new() -> Self {
         BlockManager::default()
     }
 
+    fn transaction_index_contains(
+        &self,
+        index: &IndexedBlockStore,
+        block_id: &str,
+        id: &str,
+    ) -> Result<bool, BlockManagerError> {
+        if let Some(block) = TransactionIndex::get_block_by_id(index, id)? {
+            let head = index
+                .get(&[block_id])?
+                .next()
+                .expect("BlockStore updated during transaction index check");
```

Comment:
Why is it ok to Panic here, instead of propagating an error?

Reply:
So, this had originally been part of a larger function that I broke apart. It isn't right to panic here, it is right to panic in contains_any_transaction because the persist_lock is there and the persist_lock guarantees that there was not an update to the block store.

Reply:
Actually looking at it a second time, I think it is right to panic here as the only thing to be done by the caller would be to panic. I am going to leave a note on the docstring for the function what its requirements are.